File: app/src/app/shared/utils/string.utils.ts
/**
 * Normalise une chaîne pour la comparaison insensible à la casse et aux accents.
 * Applique : lowercase → NFD (décomposition unicode) → suppression des diacritiques.
 *
 * Partagé entre `Autocomplete` et `CategorySelect` (KKS-231, RES-001).
 */
export function normalize(s: string): string {
  return s
    .toLowerCase()
    .normalize('NFD')
    .replace(/\p{Diacritic}/gu, '');
}

File: flutter/lib/src/utils/string_utils.dart
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at https://mozilla.org/MPL/2.0/.

import 'package:diacritic/diacritic.dart';

/// Normalise une chaîne pour la recherche : lowercase + suppression des
/// diacritiques (accents) + trim.
///
/// Exemple : `normalizeForSearch('Café  ')` → `'cafe'`.
///
/// Utilisé par les composants de recherche (ex: [CategorySelectExpand])
/// pour permettre des matches insensibles à la casse et aux accents.
String normalizeForSearch(String input) =>
    removeDiacritics(input.toLowerCase()).trim();
